pkg/snmp/traps/listener_test.go

```diff
@@ -26,14 +27,14 @@ func TestListenV1GenericTrap(t *testing.T) {
 	config := Config{Port: serverPort, CommunityStrings: []string{"public"}, Namespace: "totoro"}
 	Configure(t, config)
 
-	packetOutChan := make(PacketsChannel)
+	packetOutChan := make(PacketsChannel, packetsChanSize)
```

Comment:
Previously, if receivePacket() detected a timeout https://github.com/DataDog/datadog-agent/blob/main/pkg/snmp/traps/listener_test.go#L176, but the trap was subsequently received, then when the packet was written to the channel at https://github.com/DataDog/datadog-agent/blob/main/pkg/snmp/traps/listener.go#L100 it would block forever (the test would not exit as expected, until the test framework timed out after 3 minutes in CI or 10 minutes locally) because receivePacket() was no longer listening on the channel.  Making the PacketsChannels buffered avoids this issue and also mirrors the actual use https://github.com/DataDog/datadog-agent/blob/main/pkg/snmp/traps/server.go#L70.